File: src/main/java/org/zenbot/szolnok/timetable/batch/configuration/ReadUrlsTaskletStepConfiguration.java
package org.zenbot.szolnok.timetable.batch.configuration;

import org.springframework.batch.core.Step;
import org.springframework.batch.core.configuration.annotation.StepBuilderFactory;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.zenbot.szolnok.timetable.batch.batch.step.readurls.ReadFileResourcesTasklet;

@Configuration
public class ReadUrlsTaskletStepConfiguration {

    private final StepBuilderFactory stepBuilderFactory;
    private final ReadFileResourcesTasklet readFileResourcesTasklet;

    public ReadUrlsTaskletStepConfiguration(StepBuilderFactory stepBuilderFactory, ReadFileResourcesTasklet readFileResourcesTasklet) {
        this.stepBuilderFactory = stepBuilderFactory;
        this.readFileResourcesTasklet = readFileResourcesTasklet;
    }

    @Bean
    public Step readUrlsStep() {
        return stepBuilderFactory.get("readUrlsStep")
                .tasklet(readFileResourcesTasklet)
                .build();
    }
}

File: szolnok-timetable-batch-bus/src/main/kotlin/org/zenbot/szolnok/timetable/batch/bus/configuration/ReadUrlsTaskletStepConfiguration.kt
package org.zenbot.szolnok.timetable.batch.bus.configuration

import org.springframework.batch.core.Step
import org.springframework.batch.core.configuration.annotation.StepBuilderFactory
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.zenbot.szolnok.timetable.batch.utils.common.batch.tasklet.ReadFileResourcesTasklet

@Configuration
class ReadUrlsTaskletStepConfiguration(
    private val stepBuilderFactory: StepBuilderFactory,
    private val readFileResourcesTasklet: ReadFileResourcesTasklet
) {

    @Bean
    open fun readUrlsStep(): Step {
        return stepBuilderFactory.get("readUrlsStep")
                .tasklet(readFileResourcesTasklet)
                .build()
    }
}
